packages/hooks-react/src/useProfiles.ts

```diff
@@ -24,71 +21,68 @@ export const useSelectProfile = () => {
     },
     onSuccess: async () => {
       useProfileStore.setState({ selectingProfileAvatar: null });
-      navigate('/');
       await accountController?.loadUserData();
+      options?.onSuccess?.();
     },
     onError: () => {
       useProfileStore.setState({ selectingProfileAvatar: null });
-      navigate('/u/profiles');
       logDev('Unable to enter profile');
+      options?.onError?.();
     },
   });
 };
 
 export const useCreateProfile = (options?: UseMutationOptions<ServiceResponse<ProfilesData> | undefined, unknown, ProfilePayload, unknown>) => {
   const { query: listProfiles } = useProfiles();
-  const navigate = useNavigate();
 
   const profileController = getModule(ProfileController, false);
 
   return useMutation<ServiceResponse<ProfilesData> | undefined, unknown, ProfilePayload, unknown>(async (data) => profileController?.createProfile(data), {
-    onSuccess: (res) => {
-      const profile = res?.responseData;
-      if (profile?.id) {
-        listProfiles.refetch();
-        navigate(`/u/profiles?success=true&id=${profile.id}`);
-      }
-    },
     ...options,
+    onSuccess: (data, variables, context) => {
+      listProfiles.refetch();
```

Comment:
`listProfiles.refetch` is now always performed?

Reply:
On creation success indeed. Do you think it's worth it to check if the profile has an id?

Reply:
Not sure, but that is how it was before 😅

Reply:
Yes, but mostly to navigate to the correct profile page, I thought. I don't think an unnecessary refetch would occur.

Reply:
oki!